src/topp/GNPSExport.cpp

```diff
@@ -79,297 +147,368 @@ using namespace std;
+            && (int)pept_id.getMetaValue("map_index") == element_map)
+        {
+          int map_index = pept_id.getMetaValue("map_index");
+          int spec_index = pept_id.getMetaValue("spectrum_index");
+          pept_indices.emplace_back(map_index,spec_index);
+          break;
+        }
+      }
+    }
+
+    // return will be reformatted vector<PeptideIdentification> pepts passed in by value
+  }
+
+  MSExperiment& getSpectraAtIndex(const StringList& mzml_file_paths, vector<MSExperiment>& specs_list, int map_index) {
+    MSExperiment& specs = specs_list.at(map_index);
+    
+    if (specs.empty())
+    {
+      MzMLFile mzml_file;
+      mzml_file.load(mzml_file_paths[map_index], specs);
+    }
+
+    return specs_list.at(map_index);
+  }
+
 
 protected:
   // this function will be used to register the tool parameters
   // it gets automatically called on tool execution
   void registerOptionsAndFlags_() override
   {
-    registerInputFile_("in_cm", "<file>", "", "input file containing consensus elements with \'peptide\' annotations");
+    registerInputFile_("in_cm", "<file>", "", "Input consensusXML file containing only consensusElements with \"peptide\" annotations.");
     setValidFormats_("in_cm", ListUtils::create<String>("consensusXML"));
 
-    registerInputFileList_("in_mzml", "<files>", ListUtils::create<String>(""), "original mzml files containing ms/ms spectrum information");
+    registerInputFileList_("in_mzml", "<files>", ListUtils::create<String>(""), "Original mzml files containing the ms2 spectra (aka peptide annotation). \nMust be in order that the consensusXML file maps the original mzML files.");
     setValidFormats_("in_mzml", ListUtils::create<String>("mzML"));
 
     registerOutputFile_("out", "<file>", "", "Output MGF file");
     setValidFormats_("out", ListUtils::create<String>("mgf"));
 
-    registerStringOption_("output_type", "<choice>", "full_spectra", "specificity of mgf output information", false);
-    setValidStrings_("output_type", ListUtils::create<String>("full_spectra,merged_spectra"));
+    registerStringOption_("output_type", "<choice>", "most_intense", "specificity of mgf output information", false);
+    setValidStrings_("output_type", ListUtils::create<String>("merged_spectra,most_intense"));
+
+    addEmptyLine_();
 
-    registerDoubleOption_("precursor_mz_tolerance", "<num>", DEF_PRECURSOR_MZ_TOLERANCE, "Tolerance mz window for precursor selection", false);
-    registerDoubleOption_("precursor_rt_tolerance", "<num>", DEF_PRECURSOR_RT_TOLERANCE, "Tolerance rt window for precursor selection", false);
+    // registerIntOption_("msmap_cache", "<num>", DEF_MSMAP_CACHE, "Number of msmaps that can be cached during export for optimized performance", false, true);
+    registerIntOption_("peptide_cutoff", "<num>", DEF_PEPT_CUTOFF, "Number of most intense peptides to consider per consensus element; '-1' to consider all identifications", false, true);
```

Comment:
you can use `setMinInt_()` and `setMaxInt_()` for limit the allowed range of values.
E.g. -2 does not make sense here, so go for [-1, inf) (i.e. just specify the minimum)